### DISCUSSION TITLE
Custom cost function

### DISCUSSION THREAD

**Discussion by mabilinab** [valhalla/valhalla#4699] [Q&A]
Hi there!
Is there any guide or documentation on how to use a specific cost function for the routing calculation?
My cost function should look like something like this: 
for edge e linking e1 to e2
cost(edge e) = param1 * some_heuristic_distance(e2, fixed_point B) + param2 * some_probability(edge) + param3 * normal_cost(edge e)
Then I would like to route based on minimizing this specific cost.
I'm struggling to find a guide on what I should specifically change in the source code to achieve this. 
Thank you very much in advance for the help.

**Comment by nilsnolde** [comment]
You'd need to review the existing cost classes in src/sif. You can always come up with your own costing derived from those.

**Reply by mabilinab** [reply]
Thanks for the reply Nils!
Is there any available guide on how to do this? Or any code example that implements this? 
It's kinda tricky for me as I do not have C++ experience and would greatly benefit from any example.

**Reply by nilsnolde** [reply]
You’ll just need to derive a new class from sif/autocost.cc ::AutoCost and override the EdgeCost methods, then plug a few things as eg https://github.com/valhalla/valhalla/pull/4492. if that’s not enough, you can contact me on enquiry@gis-ops.com for a commercial offer.

**Reply by mabilinab** [reply]
Thank you! I'll be diving into the details very soon
